.github/workflows/main.yml

```diff
@@ -0,0 +1,42 @@
+name: Automated Tests
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        python: [3.6, 3.8, 3.9]
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install tox
+        run: |
+          python -m pip install tox
+      - name: Run tests
+        run: tox
```

Comment:
It's preferable to separate a `--notest` invocation to make the failure reasons easier to distinguish.

Reply:
In my experience, the tox output provides a suitable separation of concerns, and the added complexity is rarely worth it. If tox is not providing enough context to easily distinguish the setup phase from the run phase, then I'd say that should be escalated to tox to improve the experience. I'm disinclined to accept the burden of micro-managing the tools' experience across any number of projects. If there's a compelling story here, surely it should be applied at the tool level and not cargo-culted.

Reply:
The reason I prefer this is that I can see it separately in the CI steps. And when I open the testing step, it won't start with the `pip install` logs but rather with what actually matters. This largely improves the CI UX.
Also, in the past, there was a nasty tox bug that looked like the test invocation was failing, hiding that the real reason was the install step. Making it separate allowed to actually identify what the problem was.
Besides, this is an idiom that the tox maintainers themselves follow: https://github.com/tox-dev/azure-pipelines-template/blob/30f07f8/run-tox-env.yml#L143.

Reply:
Good discussion and good reference. Let's let this approach settle for a bit and revisit in the skeleton in the future. Feel free to file a feature request.